tss-esapi/src/structures/tagged/parameters.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright 2021 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use crate::{
+    interface_types::algorithm::PublicAlgorithm,
+    structures::{
+        PublicEccParameters, PublicKeyedHashParameters, PublicRsaParameters,
+        SymmetricCipherParameters,
+    },
+    tss2_esys::{TPMT_PUBLIC_PARMS, TPMU_PUBLIC_PARMS},
+    Error, Result,
+};
+use std::convert::{TryFrom, TryInto};
+/// Enum representing the public parameters structure.
+///
+/// # Details
+/// This corresponds to TPMT_PUBLIC_PARMS
+#[derive(Debug, Clone, Copy)]
+pub enum PublicParameters {
+    Rsa(PublicRsaParameters),
+    KeyedHash(PublicKeyedHashParameters),
+    Ecc(PublicEccParameters),
+    SymCipher(SymmetricCipherParameters),
+}
+
+impl PublicParameters {
+    /// Returns the algorithm
+    pub const fn algorithm(&self) -> PublicAlgorithm {
+        match self {
+            PublicParameters::Rsa(_) => PublicAlgorithm::Rsa,
+            PublicParameters::KeyedHash(_) => PublicAlgorithm::KeyedHash,
+            PublicParameters::Ecc(_) => PublicAlgorithm::Ecc,
+            PublicParameters::SymCipher(_) => PublicAlgorithm::SymCipher,
+        }
+    }
```

Comment:
I'm having some issues with these `const fn` methods that contain `match` statements, because they were introduced in 1.46.0 and we're trying to keep Parsec compatible with 1.43.0. There are a few methods like these scattered across the structures, my question was: do you need them as `const`? If yes, I'll change them in my PR to conditionally compile them and add non-`const` for older compiler versions.

Reply:
No, I do not need them as const. I just like to as much as possible as const (guess I am to influenced by c++ constexpr).

Reply:
It would be good to have them `const`, but alas we can't use it (for now, for Parsec) :( We can make them `const` again later when our MSRV changes, or I can try and work around it!

Reply:
I have changed them for now so you should be able to use it.